components/SidebarSection.tsx

```diff
@@ -1,60 +1,57 @@
-import React, { useEffect, useRef, useState, useCallback } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { useRouter } from 'next/router';
 import { ChevronRightIcon } from '@100mslive/react-icons';
 import { Flex, Text } from '@100mslive/react-ui';
 import SidebarItem from './SidebarItem';
 import { titleCasing } from '../lib/utils';
 
 interface Props {
-    value: String;
-    index: Number;
+    value: string;
+    index: number;
     children: any;
-    nested: Boolean;
+    nested: boolean;
 }
 
 const SidebarSection: React.FC<Props> = ({ value: key, index, children, nested = false }) => {
-    const router = useRouter() as any;
+    const router = useRouter();
     const {
         asPath,
         query: { slug }
     } = router;
-
     const activeItem = useRef<HTMLAnchorElement>(null);
+    const [openSection, setOpenSection] = useState(false);
+    const [isInFocus, setIsInFocus] = useState(false);
 
-    const isInFocus = useCallback(() => {
-        for (const i of slug) if (i === key) return true;
-        return false;
-    }, [slug, key]);
-
-    const inFocus = isInFocus();
-    const [openSection, setOpenSection] = useState(inFocus);
+    useEffect(() => {
+        if (window) {
+            // Add active sections to localStorage
+            const currentList = JSON.parse(localStorage.getItem('openedAccordions') || '[]');
+            if (openSection) {

```

Comment:
why not have this function in utils itself?

Reply:
no i meant using the localStorage function itself

Reply:
Updated